staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -157,7 +170,34 @@ func (s *SpdyRoundTripper) dial(req *http.Request) (net.Conn, error) {
 
 	rwc, _ := proxyClientConn.Hijack()
 
-	if req.URL.Scheme != "https" {
+	return s.tlsConn(req.URL, rwc, targetHost)
+}
+
+// dialWithoutProxy dials the host specified by url through a socks5 proxy.
+func (s *SpdyRoundTripper) dialWithSocks5Proxy(requestUrl *url.URL, proxyURL *url.URL) (net.Conn, error) {
+	// ensure we use a canonical host with proxyReq
+	targetHost := netutil.CanonicalAddr(requestUrl)
+
+	proxyDialAddr := netutil.CanonicalAddr(proxyURL)
+	proxyDialer, err := proxy.SOCKS5("tcp", proxyDialAddr, nil, proxy.Direct)
```

Comment:
This is a no-auth connection, are we ok with that? In fact I believe x.net.proxy only supports no-auth and username/password authentication (in the clear).

Reply:
Hey @cheftako, sorry about the little delay. I looked around a bit and I did not find any good Go library that supports authentication in a Socks5 proxy dialer. 

I think a no-auth version is still valuable as from what I understand, in most of the production-grade Kubernetes setups, the kubectl calls would already be coming from a pre-authenticated source and no authentication would be needed on the Socks5 proxy.

Reply:
@cheftako any follow up on this conversation? We've been using a k8s fork with this fix at our company and we'd love to get this merged at some point.

Reply:
You are aware that spdy is deprecated correct? (https://github.com/kubernetes/kubernetes/issues/7452)

Reply:
@mikedanese What is the current life expectancy of spdy?

Reply:
> I looked around a bit and I did not find any good Go library that supports authentication in a Socks5 proxy dialer.

This is because username/passwords are sent over clear text. The connection to the apiserver will still be e2e encrypted though, so no auth is not a major concern )(and less of a concern then password auth over cleartext).

> What is the current life expectancy of spdy?

Long :(. Even with http2, we will still want to have support for *_proxy env variables, so I would expect that we would be ok with maintaining this feature over a transition.